packages/subgraph/tests/plugin/plugin.test.ts

```diff
@@ -25,36 +64,170 @@ describe('Plugin', () => {
     clearStore();
   });
 
-  describe('NumberStored Event', () => {
-    test('it should store the correct number emitted from the event', () => {
-      const daoAddress = Address.fromString(DAO_ADDRESS);
-      const pluginAddress = Address.fromString(CONTRACT_ADDRESS);
+  describe('handleProposalCreated', () => {
+    test('test the event', () => {
+      // create state
+      createAdminPluginState(pluginEntityId);
+
+      // create event
+      let event = createNewProposalCreatedEvent(
+        PLUGIN_PROPOSAL_ID,
+        ADDRESS_ONE,
+        START_DATE,
+        START_DATE,
+        STRING_DATA,
+        actions,
+        ALLOW_FAILURE_MAP,
+        pluginEntityId
+      );
+
+      // handle event
+      _handleProposalCreated(event, DAO_ADDRESS, STRING_DATA);
+
+      let proposalEntityId = generateProposalEntityId(
+        pluginAddress,
+        BigInt.fromString(PLUGIN_PROPOSAL_ID)
+      );
 
-      const installationId = generatePluginInstallationEntityId(
-        daoAddress,
-        pluginAddress
+      // checks
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'id',
+        proposalEntityId
+      );
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'daoAddress',
+        DAO_ADDRESS
+      );
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'plugin',
+        pluginEntityId
+      );
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'pluginProposalId',
+        PLUGIN_PROPOSAL_ID
+      );
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'creator',
+        ADDRESS_ONE
       );
-      if (!installationId) {
-        throw new Error('Failed to get installationId');
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'metadata',
+        STRING_DATA
+      );
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'executed',
+        'false'
+      );
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'createdAt',
+        event.block.timestamp.toString()
+      );
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'startDate',
+        START_DATE
+      );
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'endDate',
+        START_DATE
+      );
+      assert.fieldEquals(
+        'AdminProposal',
+        proposalEntityId,
+        'allowFailureMap',
+        ALLOW_FAILURE_MAP
+      );
+
+      // check actions
+      for (let index = 0; index < actions.length; index++) {
```

Comment:
It's a bit weird to check a list of length 1 with a for loop, but I guess that's the test that existed on the subgraph.
I hope we can improve those in another task.

Reply:
Loop removed